### ISSUE TITLE
The output of forward and step are different

### ISSUE THREAD

**Issue by yjdy** [state-spaces/mamba#500]
I test forward and step function with same input_ids and intermediate states.
But the output logits and states are quite different.

The code is like this:
```python
for i in range(seqlen):
    hidden_state1 = model.step(input_ids[i],conv_state, ssm_state)

hidden_state2 = model(input_ids,inference_param)
```

I think the result should be same, but I do not know the reason.

**Comment by WorldEditors** [comment]
The same problem is found, is this problem solved?
I found there might be a bug related to the returned ssm_state (the conv_state is correct according to my investigation)

**Comment by WorldEditors** [comment]
https://github.com/state-spaces/mamba/issues/641

mark the relevant issue, still not solved
